src/Storages/FileLog/StorageFileLog.cpp

```diff
@@ -224,76 +224,48 @@ void StorageFileLog::loadFiles()
 void StorageFileLog::serialize() const
 {
     for (const auto & [inode, meta] : file_infos.meta_by_inode)
-    {
-        auto full_name = getFullMetaPath(meta.file_name);
-        if (!disk->exists(full_name))
-        {
-            disk->createFile(full_name);
-        }
-        else
-        {
-            checkOffsetIsValid(full_name, meta.last_writen_position);
-        }
-        auto out = disk->writeFile(full_name);
-        writeIntText(inode, *out);
-        writeChar('\n', *out);
-        writeIntText(meta.last_writen_position, *out);
-    }
+        serialize(inode, meta);
 }
 
 void StorageFileLog::serialize(UInt64 inode, const FileMeta & file_meta) const
 {
-    auto full_name = getFullMetaPath(file_meta.file_name);
-    if (!disk->exists(full_name))
+    auto full_path = getFullMetaPath(file_meta.file_name);
+    if (disk->exists(full_path))
     {
-        disk->createFile(full_name);
+        checkOffsetIsValid(file_meta.file_name, file_meta.last_writen_position);
     }
     else
     {
-        checkOffsetIsValid(full_name, file_meta.last_writen_position);
+        disk->createFile(full_path);
+    }
+
+    try
+    {
+        auto out = disk->writeFile(full_path);
+        writeIntText(inode, *out);
+        writeChar('\n', *out);
+        writeIntText(file_meta.last_writen_position, *out);
+    }
+    catch (...)
+    {
+        disk->removeFile(full_path);
```

Comment:
This is not reliable because the server can be killed in the middle.
We should solve it by creating a temporary file and renaming it.